### PR TITLE
Refactor Semantic Feature

### PR DESCRIPTION
**Pull request by PhoenixHe-TSP**

### CODE REVIEW

File: lib/liveValidation/liveValidator.ts [hunk 1]
```diff
@@ -70,11 +70,11 @@ export interface RequestResponseLiveValidationResult {
   readonly runtimeException?: RuntimeException;
 }
 
-export interface LiveValidationIssue extends SchemaValidateIssue {
-  readonly pathsInPayload: string[];
-  readonly severity: Severity;
-  readonly documentationUrl?: string;
-}
+export type LiveValidationIssue = {
```
Comment (raych1) [inline lib/liveValidation/liveValidator.ts:73]: Is severity not used?

File: lib/liveValidation/liveValidator.ts [hunk 2]
```diff
@@ -198,7 +198,7 @@ export class LiveValidator {
       container,
       fileRoot: this.options.directory,
       ...this.options,
-      loadSuppression: this.options.loadSuppression ?? Object.keys(allErrorConstants),
+      loadSuppression: this.options.loadSuppression ?? Object.keys(),
```
Comment (raych1) [inline lib/liveValidation/liveValidator.ts:201]: Why do we change default behavior?

Comment (PhoenixHe-msft) [reply lib/liveValidation/liveValidator.ts:201]: Unfinished. Should use another way to compute all the errorcode

File: lib/liveValidation/liveValidator.ts (hunk 1 above)
Comment (PhoenixHe-msft) [reply lib/liveValidation/liveValidator.ts:73]: It should move to lower level.

File: lib/validate.ts [hunk 3]
```diff
@@ -122,26 +121,8 @@ const prettyPrint = <T extends NodeError<T>>(
   }
 };
 
-export const validateSpec = async (
-  specPath: string,
-  options: Options | undefined
-): Promise<SpecValidationResult> =>
+export const validateSpec = async (specPath: string, options: Options | undefined) =>
   validate(options, async (o) => {
```
Comment (raych1) [inline lib/validate.ts:125]: Do we keep these options?

**Review by raych1 — APPROVED** [review]

**Pull request merged**